### ISSUE TITLE
build error when using nickel crate from github

### ISSUE THREAD

**Issue by yosemitebandit** [kaj/nickel-jwt-session#1]
I get this build error when I use the github version of nickel rather than the 0.8 release:

```
src/bin/server.rs:17:12: 17:19 error: the trait `for<'r, 'mw, 'conn> core::ops::Fn<(&'r mut nickel::request::Request<'mw, 'conn>, nickel::response::Response<'mw>)>` is not implemented for the type `nickel_jwt_session::SessionMiddleware` [E0277]
src/bin/server.rs:17     server.utilize(SessionMiddleware::new("My very secret key")
                                ^~~~~~~
src/bin/server.rs:17:12: 17:19 help: run `rustc --explain E0277` to see a detailed explanation
```

The project is the example from this repo, but with `nickel = { git = "https://github.com/nickel-org/nickel.rs" }` in `Cargo.toml`.

Any idea on how I could fix this in my own fork of nickel or nickel-jwt-session?  I unfortunately have to use this unreleased version of nickel to get access to URL-encoded form data.

**Comment by kaj** [comment]
Strange, it works for me with nickel from github.   Maybe they made an incompatible change in nickel but then reverted it?  Can you try again after a `cargo update`?

The nickel version I get is https://github.com/nickel-org/nickel.rs#653d9f55 (and yes, there are some changes that happened after you reported the bug).  They actually made a minor release as well with those changes, so `cargo update` after reverting `Cargo.toml` to say `nickel = "0.8"` should also get those changes.

**Comment by kaj** [comment]
Thanks for your interest in nickel-jwt-session!

I should probably set up travis to rebuild for every new release of nickel (and ideally, for github head as well), but I don't know how to do it or even if it is supported by travis at all.  Any hints on that?

Other than avoiding future problems, I trust your problem is solved with release 0.8.1 of nickel?

**Comment by yosemitebandit** [comment]
Hm, rebuilding when nickel updates would be cool..but I'm also not sure how to do it!

Yes, I can now build with 0.8.1 of nickel -- thanks for your help and your work on this repo!
